data/collector_types.go

```diff
@@ -0,0 +1,394 @@
+		})
+	}
+	return sd
+}
+
+// Validate returns an error if the *CaptureResult is invalid.
+func (cr *CaptureResult) Validate() error {
+	var ts Timestamps
+	if cr.Timestamps.TimeRequested == ts.TimeRequested {
+		return errors.New("Timestamps.TimeRequested must be set")
+	}
+
+	if cr.Timestamps.TimeReceived == ts.TimeReceived {
+		return errors.New("Timestamps.TimeRequested must be set")
+	}
+
+	switch cr.Type {
+	case CaptureTypeTabular:
+		if len(cr.Binaries) > 0 {
+			return errors.New("tabular result can't contain binary data")
+		}
+		if cr.TabularData.Payload == nil {
+			return errors.New("tabular result must have non empty tabular data")
+		}
+		return nil
+	case CaptureTypeBinary:
+		if cr.TabularData.Payload != nil {
+			return errors.New("binary result can't contain tabular data")
+		}
+		if len(cr.Binaries) == 0 {
+			return errors.New("binary result must have non empty binary data")
+		}
+
+		for _, b := range cr.Binaries {
+			if len(b.Payload) == 0 {
+				return errors.New("binary results can't have empty binary payload")
+			}
+		}
+		return nil
+	case CaptureTypeUnspecified:
+		return fmt.Errorf("unknown CaptureResultType: %d", cr.Type)
+	default:
+		return fmt.Errorf("unknown CaptureResultType: %d", cr.Type)
+	}
+}
+
+// CaptureType represents captured tabular or binary data.
+type CaptureType int
+
+const (
+	// CaptureTypeUnspecified represents that the data type of the captured data was not specified.
+	CaptureTypeUnspecified CaptureType = iota
+	// CaptureTypeTabular represents that the data type of the captured data is tabular.
+	CaptureTypeTabular
+	// CaptureTypeBinary represents that the data type of the captured data is binary.
+	CaptureTypeBinary
+)
+
+// ToProto converts a CaptureType into a v1.DataType.
+func (dt CaptureType) ToProto() datasyncPB.DataType {
+	switch dt {
+	case CaptureTypeTabular:
+		return datasyncPB.DataType_DATA_TYPE_TABULAR_SENSOR
+	case CaptureTypeBinary:
+		return datasyncPB.DataType_DATA_TYPE_BINARY_SENSOR
+	case CaptureTypeUnspecified:
+		return datasyncPB.DataType_DATA_TYPE_UNSPECIFIED
+	default:
+		return datasyncPB.DataType_DATA_TYPE_UNSPECIFIED
+	}
+}
+
+// GetDataType returns the DataType of the method.
+func GetDataType(methodName string) CaptureType {
+	switch methodName {
+	case nextPointCloud, readImage, pointCloudMap, GetImages, captureAllFromCamera:
+		return CaptureTypeBinary
+	default:
+		return CaptureTypeTabular
+	}
+}
+
+// Timestamps are the timestamps the data was captured.
+type Timestamps struct {
+	// TimeRequested represents the time the request for the data was started
+	TimeRequested time.Time
+	// TimeReceived represents the time the response for the request for the data
+	// was received
+	TimeReceived time.Time
+}
+
+// MimeType represents the mime type of the sensor data.
+type MimeType int
+
+// This follows the mime types supported in
+// https://github.com/viamrobotics/api/pull/571/files#diff-b77927298d8d5d5228beeea47bd0860d9b322b4f3ef45e129bc238ec17704826R75
```

Comment:
nit: link to https://github.com/viamrobotics/api/blob/d7436a969cbc03bf7e84bb456b6dbfeb51f664d7/proto/viam/app/datasync/v1/data_sync.proto#L69 instead?

Reply:
done